### DISCUSSION TITLE
Introdução a Minimal API - Dúvida sobre testar chamada api com token

### DISCUSSION THREAD

**Discussion by dalmiroe** [balta-io/forum#456] [Tira Dúvidas] [answered]
Boa tarde.
Estou na parte de autorização e gostaria de saber como fazer um teste sem ser via postman. Gostaria de fazer um teste através de uma aplicação console por exemplo para consumir a api projeto onde precise de token. Existe algum exemplo no ambiente do curso ou no site?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @dalmiroe, como vai?
Você pode utilizar o HttpClient nativo do .NET ou usar o RestSharp (https://restsharp.dev/).

```csharp
var authenticator = new JwtAuthenticator(myToken);
var options = new RestClientOptions("http://example.com") {
    Authenticator = authenticator
};
var client = new RestClient(options);
var request = new RestRequest("URL DA API");
var response = await client.GetAsync(request, cancellationToken);
```

**Comment by dalmiroe** [comment]
Funcionou das duas formas
Obrigado 

 

De: Andre Baltieri ***@***.***> 
Enviada em: sexta-feira, 5 de janeiro de 2024 19:11
Para: balta-io/forum ***@***.***>
Cc: Dalmiro ***@***.***>; Mention ***@***.***>
Assunto: Re: [balta-io/forum] Introdução a Minimal API - Dúvida sobre testar chamada api com token (Discussion #456)

 

Boa tarde, @dalmiroe <https://github.com/dalmiroe> , como vai?
Você pode utilizar o HttpClient nativo do .NET ou usar o RestSharp (https://restsharp.dev/).

var authenticator = new JwtAuthenticator(myToken);
var options = new RestClientOptions("http://example.com") {
    Authenticator = authenticator
};
var client = new RestClient(options);
var request = new RestRequest("URL DA API");
var response = await client.GetAsync(request, cancellationToken);
